dev/src/serializer.ts

```diff
@@ -116,17 +116,16 @@ export class Serializer {
       };
     }
 
-    if (typeof val === 'number' && !isNaN(val) && val % 1 === 0) {
-      return {
-        integerValue: val as number,
-      };
-    }
-
-    // Integers are handled above, the remaining numbers are treated as doubles
     if (typeof val === 'number') {
-      return {
-        doubleValue: val as number,
-      };
+      if (isSafeInteger(val)) {
```

Comment:
You can directly use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger since we are using ES2016: https://github.com/google/gts/blob/master/tsconfig-google.json#L7